client/landing/gutenboarding/components/duplicate-site-modal/index.tsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect, useState } from 'react';
+import { Button, Modal } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { useI18n } from '@automattic/react-i18n';
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY as ONBOARD_STORE } from '../../stores/onboard';
+import { SITE_STORE } from '../../stores/site';
+import './style.scss';
+
+interface Props {
+	onRequestClose: () => void;
+}
+
+const DuplicateSiteModal = ( { onRequestClose }: Props ) => {
+	const { __: NO__ } = useI18n();
+
+	const [ showDuplicateSiteModal, setShowDuplicateSiteModal ] = useState( false );
+
+	const lastCreatedSite = useSelect( select => select( ONBOARD_STORE ).getLastCreatedSite() );
+	const existingSite = useSelect( select => {
+		return select( SITE_STORE ).getExistingSite( lastCreatedSite?.domain );
+	} );
+
+	useEffect( () => {
+		recordTracksEvent( 'calypso_duplicate_site_modal_start', {
+			flow: 'gutenboarding',
+		} );
+	}, [] );
+
+	useEffect( () => {
+		if ( existingSite ) {
+			setShowDuplicateSiteModal( true );
+		}
```

Comment:
Thanks! Yes, you're right it doesn't need to be here at all. I think I was torn about where to put the logic and then got stuck half way in between two approaches. I'll have a go at moving the check to `gutenboarding.tsx` and pass down the existing site in props so that the modal doesn't have to be aware of state (this conditional was originally there to prevent the modal from rendering empty strings if the details of the last created site weren't resolved yet).